pkg/auth/dependency/userprofile/tx_provider.go

```diff
@@ -22,12 +23,17 @@ func NewSafeProvider(
 	}
 }
 
-func (s *safeUserProfileImpl) CreateUserProfile(userID string, data Data) (profile UserProfile, err error) {
+func (s *safeUserProfileImpl) CreateUserProfile(userID string, accessToken string, data Data) (profile UserProfile, err error) {
 	s.txContext.EnsureTx()
 	return s.impl.CreateUserProfile(userID, data)
 }
 
-func (s *safeUserProfileImpl) GetUserProfile(userID string) (profile UserProfile, err error) {
+func (s *safeUserProfileImpl) GetUserProfile(userID string, accessToken string) (profile UserProfile, err error) {
 	s.txContext.EnsureTx()
 	return s.impl.GetUserProfile(userID)
 }
+
+func (s *safeUserProfileImpl) WithInTx() (withIn bool) {
```

Comment:
I thought `WithInTx` was saying if the store is now in transaction.

I suggest to use a function name that describe the store characteristics, something like `IsOperationRunInDB` or `CanWithInTx`.